pkg/cmd/pulumi/schema_generate_sdk.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2016-2021, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/spf13/cobra"
+
+	dotnetgen "github.com/pulumi/pulumi/pkg/v3/codegen/dotnet"
+	gogen "github.com/pulumi/pulumi/pkg/v3/codegen/go"
+	nodejsgen "github.com/pulumi/pulumi/pkg/v3/codegen/nodejs"
+	pythongen "github.com/pulumi/pulumi/pkg/v3/codegen/python"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+func newSchemaGenerateSDKCommand() *cobra.Command {
+	var target string
+	var outputDirectory string
+
+	cmd := &cobra.Command{
+		Use:   "generate-sdk",
+		Args:  cmdutil.ExactArgs(1),
+		Short: "Generate language SDKs from a Pulumi package schema",
+		Long: "Generate language SDKs from a Pulumi package schema.\n" +
+			"\n" +
+			"Given a Pulumi package schema, write out language SDKs for the package.\n" +
+			"The schema is validated as if by `pulumi schema check` prior to generation.\n" +
+			"\n" +
+			"Valid targets are dotnet, go, nodejs, and python. The 'all' target will\n" +
+			"generate an SDK for each suported target.",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			if outputDirectory == "" {
+				return fmt.Errorf("an output directory must be specified")
+			}
+
+			pkgSpec, err := loadSchemaSpec(args[0], "")
+			if err != nil {
+				return err
+			}
+
+			pkg, diags, err := schema.BindSpec(pkgSpec, nil)
+			if len(diags) != 0 {
+				diagWriter := hcl.NewDiagnosticTextWriter(os.Stderr, nil, 0, true)
+				wrErr := diagWriter.WriteDiagnostics(diags)
+				contract.IgnoreError(wrErr)
+
+				if diags.HasErrors() {
+					return fmt.Errorf("too many errors")
```

Comment:
I don't love `"too many errors"`. To me, it implies that some amount of errors are acceptable. It makes me think of there being too many errors to display. I would say
```suggestion
                                      return fmt.Errorf("failed to generate %v", args[0])
```
```